src/com/palmergames/bukkit/towny/command/TownCommand.java

```diff
@@ -1887,34 +1892,69 @@ private static void parseJailCommand(CommandSender sender, Town town, String[] s
 					try {
 						Integer.parseInt(split[1]);
 						if (split.length > 2)
-							Integer.parseInt(split[2]);
+							Double.parseDouble(split[2]);
 						if (split.length > 3)
 							Integer.parseInt(split[3]);
+						if (split.length > 4)
+							Integer.parseInt(split[4]);
 					} catch (NumberFormatException e) {
 						HelpMenu.TOWN_JAIL.send(sender);
 						return;
 					}
+					
 					hours = Integer.valueOf(split[1]);
-					if (hours < 1)
+					if (hours < 1) 
 						hours = 1;
-	
+					if (hours > TownySettings.getJailedMaxSetHours())
+					{
+						hours = TownySettings.getJailedMaxSetHours();
+						TownyMessaging.sendMsg(sender, Translatable.of("msg_err_higher_than_max_allowed_hours_x", TownySettings.getJailedOutlawJailHours()));
+					}
+
 					if (split.length >= 3) {
-						jailNum = Integer.valueOf(split[2]);
+						bail = Double.valueOf(split[2]);
+						if (bail < 1)
+							bail = 1;
+						if (bail > TownySettings.getBailMaxAmount())
+						{
+							bail = TownySettings.getBailMaxAmount();
+							TownyMessaging.sendMsg(sender, Translatable.of("msg_err_higher_than_max_allowed_bail_x", TownySettings.getBailMaxAmount()));
+						}
+					}
+					if (split.length == 4) {
+						jailNum = Integer.valueOf(split[3]);
 						jail = town.getJail(jailNum);
 						if (jail == null) 
 							throw new TownyException(Translatable.of("msg_err_the_town_does_not_have_that_many_jails"));
 					}
 
-					if (split.length == 4) {
-						cell = Integer.valueOf(split[3]) - 1;
+					if (split.length == 5) {
+						cell = Integer.valueOf(split[4]) - 1;
 						if (!jail.hasJailCell(cell))
 							throw new TownyException(Translatable.of("msg_err_that_jail_plot_does_not_have_that_many_cells"));
 					}
 				}
 
-				JailUtil.jailResident(jailedResident, jail, cell, hours, JailReason.MAYOR, sender);
+				// Check if Town has reached max potential jailed and react according to maxJailedNewJailBehavior in config, sanitise for between 0-2
+				if (TownySettings.getMaxJailedPlayerCount() >= 0 && TownySettings.getMaxJailedPlayerCount() <= 2  && town.getJailedPlayerCount(town.getJailed()) >= TownySettings.getMaxJailedPlayerCount())

```

Comment:
I might be missing something but I don't see how `TownySettings.getMaxJailedPlayerCount() <= 2 &&` fits into this line, maybe it was left over in testing? The comment for the if statement also talks about sanitizing for 0-2 but I don't see why.

Reply:
Code has been simplified in new commit to simply check if greater than 0.

Reply:
It's currently set to greater than or equal to 0, is there ever a situation where a server would want to have MaxJailedPlayerCount at 0?

Reply:
My apologies just realised I had mentally conflated this part of the code with the feature for maxJailedNewJailBehavior hence the original weird contribution.
Agreed regarding no need for a MaxJailedPlayerCount at 0, set to check for greater than or equal to 1 as opposed to 0.